src/components/plots/abstractPlot.ts

```diff
@@ -311,37 +311,34 @@ export module Plot {
     }
 
     /**
-     * Removes a dataset by string key
+     * Removes a dataset by the given identifier
      *
-     * @param {string} key The key of the dataset
-     * @return {Plot} The calling Plot.
+     * @param {string | Dataset | any[]} datasetIdentifer The identifier for the dataset to remove
+     * If a string is inputted, it is interpreted as the dataset key to remove.
+     * If a Dataset is inputted, the first Dataset in the plot that is the same will be removed.
+     * If an any[] is inputted, the first data array in the plot that is the same will be removed.
```

Comment:
Generally, maybe "If ____, [...]" is clearer than "If a ______ is inputted, [...]"